### PR TITLE
Switch to python 3

### PR DESCRIPTION
**Pull request by andreadelprete**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -65,7 +65,7 @@ IF(NOT CLIENT_ONLY)
 
   SET(BOOST_COMPONENTS system thread regex)
   SEARCH_FOR_BOOST ()
-  FINDPYTHON(2.7 EXACT)
+  FINDPYTHON(3 REQUIRED)
```
Comment (nim65s) [inline CMakeLists.txt]: This will break python 2. Did you tried with `FINDPYTHON()` ?

Comment (andreadelprete) [reply CMakeLists.txt]: I have both python 2 and python 3 installed. What about having a separate branch of gepetto-viewer-corba for python 3? Or allowing the user to choose the python version with a CMake variable?

Comment (nim65s) [reply CMakeLists.txt]: If you use only `FINDPYTHON()`, you can already choose the python version you want at compilation time with CMake variables. 

For example:
```
-DPYTHON_EXECUTABLE=${PYTHON}
-DPYTHON_INCLUDE_DIR=${PYTHON_INCLUDE}
-DPYTHON_LIBRARY=${PYTHON_LIB}
```

I am not sure that you need all of those, the first might be enough.

Comment (jmirabel) [reply CMakeLists.txt]: That is a bit tedious to type. I would prefer a CMake variable like `PYTHON_VERSION`.

**Comment by jmirabel** [comment]
Could you add a section in the README with the few steps you did to make it work ?

File: CMakeLists.txt (hunk 1 above)
Comment (nim65s) [reply CMakeLists.txt]: for reference: https://github.com/nim65s/robotpkg/blob/master/mk/sysdep/python.mk#L90

Comment (nim65s) [reply CMakeLists.txt]: @jmirabel yes, that would be better, but for this I suggest to open an issue / PR in the cmake submodule

Comment (nim65s) [reply CMakeLists.txt]: So I tried with just `-DPYTHON_EXECUTABLE=/usr/bin/python3.5` on 16.04, and it worked:
```
-- Found PythonInterp: /usr/bin/python3.5 (found version "3.5.2") 
-- PythonInterp: /usr/bin/python3.5
-- Found PythonLibs: /usr/lib/x86_64-linux-gnu/libpython3.5m.so (found version "3.5.2") 
-- PythonLibraries: /usr/lib/x86_64-linux-gnu/libpython3.5m.so
-- PythonLibraryDirs: /usr/lib/x86_64-linux-gnu
-- PythonLibVersionString: 3.5.2
```

Comment (jmirabel) [reply CMakeLists.txt]: I am fine with `PYTHON_EXECUTABLE` if we document it somewhere.

Comment (nim65s) [reply CMakeLists.txt]: we also have to remove `FINDPYTHON` in `src/CMakeLists.txt`

Comment (jmirabel) [reply CMakeLists.txt]: I tried `-DPYTHON_EXECUTABLE=/usr/bin/python3.5` and I get:
```
-- Found PythonInterp: /usr/bin/python3 (found version "3.5.2") 
-- PythonInterp: /usr/bin/python3
-- Found PythonLibs: /usr/lib/x86_64-linux-gnu/libpython2.7.so (found version "2.7.12") 
-- PythonLibraries: /usr/lib/x86_64-linux-gnu/libpython2.7.so
-- PythonLibraryDirs: /usr/lib/x86_64-linux-gnu
-- PythonLibVersionString: 2.7.12
```
`libpython3.5` is installed on my computer but not `libpython3.5-dev`. Anyway, I should get an error and I don't.

Comment (nim65s) [reply CMakeLists.txt]: @jmirabel can you try with https://github.com/nim65s/jrl-cmakemodules/commit/bed90c47ef3fe868f17e1d8f01384da117c1ebe5 ? :)

**Comment by andreadelprete** [comment]
> Could you add a section in the README with the few steps you did to make it work ?

Done!

**Comment by nim65s** [comment]
@andreadelprete can you switch to `FINDPYTHON()` in `CMakeLists.txt` and remove the `FINDPYTHON` in `src/CMakeLists.txt` ?

**Comment by andreadelprete** [comment]
> `FINDPYTHON`

Done!

**Comment by nim65s** [comment]
This looks good to me, thanks a lot !

**Comment by jmirabel** [comment]
Thanks !

**Comment by andreadelprete** [comment]
Thanks to you guys for the help!

**Comment by andreadelprete** [comment]
Do you plan to propagate these changes to robotpkg so that python 3 bindings will be available without even there?

**Comment by jmirabel** [comment]
I didn't see the merge was done on master. This may cause build error on downstream dependencies.

I'll push it to devel and put master where it was. I'll be included in the next release.

**Comment by nim65s** [comment]
I opened a PR on robotpkg to update omniorb to 4.2.2: https://git.openrobots.org/issues/184
This is a first step toward getting this work, but we also need to enable the compilation / installation of the python3 files from omniORB / omniORBpy in robotpkg's binary packages.

**Pull request merged**